Lombiq.Tests.UI/Services/UITestExecutionSession.cs

```diff
@@ -68,6 +69,8 @@ public async Task<bool> ExecuteAsync(int retryCount, string dumpRootPath)
             var setupConfiguration = _configuration.SetupConfiguration;
             _hasSetupOperation = setupConfiguration.SetupOperation != null;
 
+            _IsDBSetupSnapshotDirectoryPath = setupConfiguration.SetupSnapshotDirectoryPath == "AppFolder";
```

Comment:
You can change the logic here to load the snapshot if the folder exists and there's no setup operation.

Reply:
You mean folder as "AppFolder" right? Because if we check if the `setupConfiguration.SetupSnapshotDirectoryPath`  exists it will be true with `Temp\SetupSnapshot` too, so it will cause the problem that I mentioned above.

Reply:
I meant `SetupSnapshotDirectoryPath`. You can also check if it's not empty.

Reply:
`Temp\SetupSnapshot` is not empty it has a subfolder, so that's not enough. So I think  `Directory.Exists(setupConfiguration.SetupSnapshotDirectoryPath + "\\App_Data")` would be ok. What do you think? (I mean it's working that way, I'm just asking if it's okay. To avoid review ping-ponging)

Reply:
Actually, loading the DB as a snapshot can be a setup operation, no? It's also applicable semantically And then no such check is necessary.

If not, what you've written sounds OK too.

Reply:
So, it can't be a setup operation?

Reply:
Yeah, if it has setup operation it does things that will break the test in case of a custom snapshot (like it combines setup snapshot directory path, with other things),

Reply:
OK then!